x-pack/packages/kbn-elastic-assistant/impl/assistant_context/assistant_nav_link.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { createHtmlPortalNode, OutPortal, InPortal } from 'react-reverse-portal';
+import { EuiToolTip, EuiButton, EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useAssistantContext } from '.';
+import { AssistantAvatar } from '../..';
+import { ChromeStyle } from '@kbn/core-chrome-browser';
+
+const isMac = navigator.platform.toLowerCase().indexOf('mac') >= 0;
+
+const TOOLTIP_CONTENT = i18n.translate(
+  'xpack.elasticAssistant.assistantContext.assistantNavLinkShortcutTooltip',
+  {
+    values: { keyboardShortcut: isMac ? '⌘ ;' : 'Ctrl ;' },
+    defaultMessage: 'Keyboard shortcut {keyboardShortcut}',
+  }
+);
+const LINK_LABEL = i18n.translate('xpack.elasticAssistant.assistantContext.assistantNavLink', {
+  defaultMessage: 'AI Assistant',
+});
+
+export const AssistantNavLink: FC = () => {
+  const {
+    chrome,
+    showAssistantOverlay,
+    assistantAvailability,
+  } = useAssistantContext();
+  const portalNode = React.useMemo(() => createHtmlPortalNode(), []);
+  const [chromeStyle, setChromeStyle] = useState<ChromeStyle | undefined>(undefined);
+
+  // useObserverable would change the order of re-renders that are tested against closely. 
```

Comment:
It would make sense to use useObserverable to observe the `chrome.getChromeStyle$()` observable. The problem is that useObserverable uses `useLayoutEffect` internally ([see here](https://github.com/streamich/react-use/commit/376eea3)) and useLayoutEffect is changing the number and order of hooks being rendered causing the following tests to fail:
<details>
  <summary>Failing tests</summary>
 ## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/259711)
* Commit: 21e861380880dab9cd29595bc2cb7a4468bd8a13

### Failed CI Steps
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2)
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6)
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4504-a425-09edd7850bcd)
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-d79e-411f-bc3a-6cc22bcd0255)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4504-a425-09edd7850bcd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4504-a425-09edd7850bcd/artifacts/0193bade-bafc-481c-b173-32954aef8ccb) Jest Tests #<span></span>15 / useAggregatedAnomaliesByJob does not throw error when aggregations is undefined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-d79e-411f-bc3a-6cc22bcd0255) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-d79e-411f-bc3a-6cc22bcd0255/artifacts/0193baf1-0552-47a8-a34e-9ee9ad104c93) Jest Tests #<span></span>15 / useAggregatedAnomaliesByJob does not throw error when aggregations is undefined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4504-a425-09edd7850bcd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4504-a425-09edd7850bcd/artifacts/0193bade-bafc-49bb-a27d-04bf8dd295de) Jest Tests #<span></span>15 / useAggregatedAnomaliesByJob returns formated data
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-d79e-411f-bc3a-6cc22bcd0255) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-d79e-411f-bc3a-6cc22bcd0255/artifacts/0193baf1-0551-44f4-aaf5-d0ad4c0aeaae) Jest Tests #<span></span>15 / useAggregatedAnomaliesByJob returns formated data
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4504-a425-09edd7850bcd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4504-a425-09edd7850bcd/artifacts/0193bade-bafb-4c6f-938b-0bfda145dbe2) Jest Tests #<span></span>15 / useAggregatedAnomaliesByJob returns jobs sorted by name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-d79e-411f-bc3a-6cc22bcd0255) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-d79e-411f-bc3a-6cc22bcd0255/artifacts/0193baf1-0551-4d08-bf42-88834e6cff59) Jest Tests #<span></span>15 / useAggregatedAnomaliesByJob returns jobs sorted by name
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd7-44f0-a8ce-b44273cb4e44) Jest Tests #<span></span>13 / useSignalIndex fetch alerts info
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf47-4930-9f5c-1aef5d24be0b) Jest Tests #<span></span>13 / useSignalIndex fetch alerts info
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd8-40bd-9926-021b3a5d88f5) Jest Tests #<span></span>13 / useSignalIndex if there is an error during createSignalIndex, we should get back signalIndexExists === false && signalIndexName == null
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf47-4d67-8c20-22fcde40f150) Jest Tests #<span></span>13 / useSignalIndex if there is an error during createSignalIndex, we should get back signalIndexExists === false && signalIndexName == null
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd3-4650-9836-90333ce29594) Jest Tests #<span></span>13 / useSignalIndex if there is an error when fetching alerts info, signalIndexExists === false && signalIndexName == null
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf48-4bf4-8e26-2a68e1faadf3) Jest Tests #<span></span>13 / useSignalIndex if there is an error when fetching alerts info, signalIndexExists === false && signalIndexName == null
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd3-4de5-b0d7-da220d31c26c) Jest Tests #<span></span>13 / useSignalIndex init
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf44-48b6-98b8-a66352384220) Jest Tests #<span></span>13 / useSignalIndex init
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd6-4745-a620-76adca8b8125) Jest Tests #<span></span>13 / useSignalIndex make sure that createSignalIndex have been called when trying to create signal index
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf48-48a0-8f00-dfbc2e67c133) Jest Tests #<span></span>13 / useSignalIndex make sure that createSignalIndex have been called when trying to create signal index
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd9-457f-8aca-2b501401b7fd) Jest Tests #<span></span>13 / useSignalIndex make sure that createSignalIndex is giving back the signal info
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf47-417f-bbc7-102d99732803) Jest Tests #<span></span>13 / useSignalIndex make sure that createSignalIndex is giving back the signal info
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd4-4dee-91f3-d8344b1fe086) Jest Tests #<span></span>13 / useSignalIndex should not make API calls when signal index already stored in sourcerer
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf46-4011-bbb5-f0037225debe) Jest Tests #<span></span>13 / useSignalIndex should not make API calls when signal index already stored in sourcerer
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193babc-c68e-4974-854b-5791c5e0cbe2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193babc-c68e-4974-854b-5791c5e0cbe2/artifacts/0193bade-7cd4-4042-a73c-1b098c3e20d7) Jest Tests #<span></span>13 / useUserInfo returns default state
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/259711#0193bade-91d7-4ecf-98e3-e164b9ef71b6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/259711/jobs/0193bade-91d7-4ecf-98e3-e164b9ef71b6/artifacts/0193baea-bf45-48d1-a5d2-861a03ae6e64) Jest Tests #<span></span>13 / useUserInfo returns default state

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/50068fd9bae6791594bd9d25cfef1f07d2b5ceab) | [after](https://github.com/elastic/kibana/commit/21e861380880dab9cd29595bc2cb7a4468bd8a13) | diff |
| --- | --- | --- | --- |
| `integrationAssistant` | 473 | 475 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/50068fd9bae6791594bd9d25cfef1f07d2b5ceab) | [after](https://github.com/elastic/kibana/commit/21e861380880dab9cd29595bc2cb7a4468bd8a13) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 14.7MB | 14.7MB | +8.7KB |



### History
* :broken_heart: [Build #259671](https://buildkite.com/elastic/kibana-pull-request/builds/259671) failed e731e5a43e22e42f06aa0622d48f49601f0aa86d
* :broken_heart: [Build #259641](https://buildkite.com/elastic/kibana-pull-request/builds/259641) failed 08b8d0f4d58debf374565f3ad49edb7452ef95bc
* :broken_heart: [Build #259572](https://buildkite.com/elastic/kibana-pull-request/builds/259572) failed 8092bedc33fe98c08bd0fbb5bef41ddc647bac54
* :broken_heart: [Build #259570](https://buildkite.com/elastic/kibana-pull-request/builds/259570) failed c2925c35269a10a5aff6bdf266c0795703bba3b0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259711","number":259711,"commit":"21e861380880dab9cd29595bc2cb7a4468bd8a13"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259671","number":259671,"commit":"e731e5a43e22e42f06aa0622d48f49601f0aa86d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259641","number":259641,"commit":"08b8d0f4d58debf374565f3ad49edb7452ef95bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259572","number":259572,"commit":"8092bedc33fe98c08bd0fbb5bef41ddc647bac54"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/259570","number":259570,"commit":"c2925c35269a10a5aff6bdf266c0795703bba3b0"}],"number":259711}
buildkite-pr-comment-->
</details>